tests/postgresql/test-postgresql.cpp

```diff
@@ -990,6 +990,71 @@ TEST_CASE("Bulk iterators", "[postgresql][bulkiters]")
     sql << "drop table t";
 }
 
+
+// false_bind_variable_inside_identifier
+struct test_false_bind_variable_inside_identifier_table_creator : table_creator_base
+{
+    test_false_bind_variable_inside_identifier_table_creator(session & sql)
+        : table_creator_base(sql)
+        , msession(sql) 
+    {
+
+        try
+        {
+            sql << "CREATE TABLE soci_test( \"column_with:colon\" integer)";
+            sql << "CREATE TYPE \"type_with:colon\" AS ENUM ('en_one', 'en_two');";
+            sql <<  "CREATE FUNCTION \"function_with:colon\"() RETURNS integer LANGUAGE 'sql' AS "
+                    "$BODY$"
+                    "   SELECT \"column_with:colon\" FROM soci_test LIMIT 1; "
+                    "$BODY$;"
+            ;
+        }
+        catch(...)
+        {
+            drop();
+        }
+        
+    }
+    ~test_false_bind_variable_inside_identifier_table_creator(){
+        drop();
+    }
+private:
+    void drop()
+    {
+        try
+        {
+            msession << "DROP FUNCTION IF EXISTS \"function_with:colon\"();";
+            msession << "DROP TYPE IF EXISTS \"type_with:colon\" ;";
+        }
+        catch (soci_error const& e){}
+    }
+    session& msession;
+};
+TEST_CASE("false_bind_variable_inside_identifier", "[postgresql][bind-variables]")
+{
+    std::string col_name;
+    int fct_return_value;
+    std::string type_value;
+
+    {
+        session sql(backEnd, connectString);
+        test_false_bind_variable_inside_identifier_table_creator tableCreator(sql);
+
+        try
+        {
+            sql << "insert into soci_test(\"column_with:colon\") values(2020)";
+            sql << "SELECT column_name FROM information_schema.columns WHERE table_schema = current_schema() AND table_name = 'soci_test';", into(col_name);
+            sql << "SELECT \"function_with:colon\"() ;", into(fct_return_value);
+            sql << "SELECT unnest(enum_range(NULL::\"type_with:colon\"))  ORDER BY 1 LIMIT 1;", into(type_value);
+        }
+        catch(...){}
```

Comment:
Why do we do this? Is the test code supposed to throw anything? Because otherwise we really should consider an exception a failure in the test instead of just ignoring them.

Reply:
It was not really mandatory.
removed.